compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java

```diff
@@ -3014,6 +3015,47 @@ private boolean inferredIsolated(BInvokableSymbol symbol, IsolationInferenceInfo
         return true;
     }
 
+    private void logServiceIsolationWarnings(List<BLangClassDefinition> classDefinitions) {
```

Comment:
This would currently result in warnings even in scenarios where the spec says a service/method will be inferred as isolated but the implementation doesn't infer so yet.